main.tf

```diff
@@ -1,13 +1,103 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled          = module.this.enabled
+  iam_role_enabled = local.enabled && var.iam_role_enabled
+}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
-  }
+module "eks_iam_policy" {
+  source  = "cloudposse/iam-policy/aws"
+  version = "0.1.0"
+
+  enabled = local.iam_role_enabled
+
+  iam_source_json_url   = var.iam_source_json_url
+  iam_policy_statements = var.iam_policy_statements
+
+  context = module.this.context
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+module "eks_iam_role" {
+  source  = "cloudposse/eks-iam-role/aws"
+  version = "0.8.0"
+
+  enabled = local.iam_role_enabled
+
+  aws_account_number          = var.aws_account_number
```

Comment:
See comment regarding`aws` provider and `aws_partition` data source.

Perhaps we should make the `aws` provider required and just use the [aws_caller_identity data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity).

Reply:
The eks iam role module has an open PR for this I believe.